pkg/controllers/resources/ingressclasses/syncer.go

```diff
@@ -71,6 +72,11 @@ func (i *ingressClassSyncer) SyncToVirtual(ctx *synccontext.SyncContext, event *
 }
 
 func (i *ingressClassSyncer) Sync(ctx *synccontext.SyncContext, event *synccontext.SyncEvent[*networkingv1.IngressClass]) (_ ctrl.Result, retErr error) {
+	// If the virtual object has the name annotation, it means it was created by vCluster and we can safely manage it.
+	if _, ok := event.Virtual.GetAnnotations()[translate.HostNameAnnotation]; !ok {
```

Comment:
1-2 months ago we decided not to use annotations and labels in this way, where annotations and labels impact the behavior (I made a similar change, maybe even smaller), IIRC because users can make changes to those and it's not secure.

cc @FabianKramm can you take a look here?